### PR TITLE
CLOUDP-106124: [mongocli] Implement mongocli auth whoami

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-106124

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

```bash
./bin/mongocli auth login --profile dev --loginOnly

First, copy your one-time code: 6QJ9-QBPH

Next, sign with your browser and enter the code.

Or go to https://account-dev.mongodb.com/account/connect

Your code will expire after 10 minutes.
Successfully logged in as redacted@mongodb.com.

./bin/mongocli auth whoami --profile dev
Logged in as redacted@mongodb.com

./bin/mongocli auth logout --profile dev
? Are you sure you want to log out of account redacted@mongodb.com? Yes
Successfully logged out of account redacted@mongodb.com

./bin/mongocli auth whoami --profile dev
Error: not logged in
```

### CODE REVIEW

**Review by blva — APPROVED** [review]
only questions

File: internal/config/profile.go [hunk 1]
```diff
@@ -263,6 +264,15 @@ func (p *Profile) AuthToken() string {
 	return p.GetString(authToken)
 }
 
+// Access will return a jwt.Token and jwt.Standard claims, this method is not safe to verify the token is valid.
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/config/profile.go]: this is about JWT being self signed and signature verification, we don't need to verify signature at this point I just wanted to get the claims so it should be safe, I updated the comment, let me know what you think

File: internal/cli/auth/whoami.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	OutWriter io.Writer
+	account   string
+}
+
+func (opts *whoOpts) Run() error {
+	_, _ = fmt.Fprintf(opts.OutWriter, "Logged in as %s\n", opts.account)
+
+	return nil
+}
+
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/auth/whoami.go:37]: not anymore, things are changing 😶

File: internal/config/profile.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (blva) [reply internal/config/profile.go]: cool, thanks! Looks good.

**Review by blva — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/cli/auth/login.go [hunk 3]
```diff
@@ -51,7 +57,7 @@ type loginOpts struct {
 	isCloudManager bool
 	noBrowser      bool
 	loginOnly      bool
-	config         config.SetSaver
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/auth/login.go:54]: if you feel strong about this feel free to enable this https://github.com/golangci/golangci-lint/blob/c270e521aa5809ab929218185ba786f5fb9040fd/.golangci.example.yml#L740 
There's no value on having to remember this

**Pull request merged**